src/utilities/convex_hull.zig

```diff
@@ -674,45 +720,98 @@ test "Merge convex hulls with random elements" {
     defer convex_hull_one.deinit();
 
     for (0..num_points) |i| {
-        try convex_hull_one.addPoint(.{ .time = i, .value = rnd.random().float(f64) });
+        try convex_hull_one.add(.{ .time = i, .value = rnd.random().float(f64) });
     }
 
     var convex_hull_two = try ConvexHull.init(allocator);
     defer convex_hull_two.deinit();
 
     for (num_points..2 * num_points) |i| {
-        try convex_hull_two.addPoint(.{ .time = i, .value = rnd.random().float(f64) });
+        try convex_hull_two.add(.{ .time = i, .value = rnd.random().float(f64) });
     }
 
     // Merge convex_hull_two into convex_hull_one.
-    try convex_hull_one.merge(&convex_hull_two);
+    try convex_hull_one.merge(&convex_hull_two, null);
 
     try testConvexHullProperty(&convex_hull_one);
 }
 
-test "Merge single element's convex hull with other convex hull" {
+test "Merge not-in-place convex hulls with random elements" {
     const num_points: usize = 100;
     const allocator = testing.allocator;
     var rnd = std.rand.DefaultPrng.init(0);
 
     var convex_hull_one = try ConvexHull.init(allocator);
     defer convex_hull_one.deinit();
 
-    try convex_hull_one.addPoint(.{ .time = 0, .value = rnd.random().float(f64) });
+    for (0..num_points) |i| {
+        try convex_hull_one.add(.{ .time = i, .value = rnd.random().float(f64) });
+    }
+
+    var convex_hull_two = try ConvexHull.init(allocator);
+    defer convex_hull_two.deinit();
+
+    for (num_points..2 * num_points) |i| {
+        try convex_hull_two.add(.{ .time = i, .value = rnd.random().float(f64) });
+    }
+
+    var convex_hull_merged = try ConvexHull.init(allocator);
+    defer convex_hull_merged.deinit();
+
+    // Merge convex_hull_two into convex_hull_one.
+    try convex_hull_one.merge(&convex_hull_two, &convex_hull_merged);
+
+    try testConvexHullProperty(&convex_hull_merged);
+}
+
+test "Merge in-place single element's convex hull with other convex hull" {
+    const num_points: usize = 100;
+    const allocator = testing.allocator;
+    var rnd = std.rand.DefaultPrng.init(0);
+
+    var convex_hull_one = try ConvexHull.init(allocator);
+    defer convex_hull_one.deinit();
+
+    try convex_hull_one.add(.{ .time = 0, .value = rnd.random().float(f64) });
 
     var convex_hull_two = try ConvexHull.init(allocator);
     defer convex_hull_two.deinit();
 
     for (1..num_points) |i| {
-        try convex_hull_two.addPoint(.{ .time = i, .value = rnd.random().float(f64) });
+        try convex_hull_two.add(.{ .time = i, .value = rnd.random().float(f64) });
     }
 
-    try convex_hull_one.merge(&convex_hull_two);
+    try convex_hull_one.merge(&convex_hull_two, null);
 
     try testConvexHullProperty(&convex_hull_one);
 }
 
-test "Merge convex hull with single element's convex hull" {
+test "Merge not-in-place single element's convex hull with other convex hull" {
```

Comment:
Seems very similar tests for in_place and not_in_place of the convex hull are required, maybe make helper functions that performs each test and then call it with and without the last argument in the actual tests to reduce code duplication.

Reply:
Done.